Automattic/wp-calypso
# Add tracks analytics to gutenboarding steps

Part of https://github.com/Automattic/wp-calypso/issues/41059

This adds tracks calls on entering and leaving gutenboarding steps.
It currently handles the first four steps:
`IntentGathering`, `DesignSelection`, `Style`, `Signup`

They don't exactly match what was in the diagram but are based on the existing type names in code.

The main complication was that the signup modal is open over the style page. 

### Testing instructions
The enter event should be fired when entering a page.
The leave event  should be fired when progressing to the next or previous page
The leave event should be fired when using browser refresh or typing into the url bar
The leave event for Style should be fired when opening the signup modal
The leave event for Style should not be fired when completing signup, or navigating away from the signup modal.

### Known issues
When opening the signup modal, the `calypso_signup_step_enter, { signup}` event is triggered slightly before the `calypso_signup_leave, { style }` event. I don't think this matters.
